libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -447,10 +448,25 @@ def _wait_for_webui(self):
                     # Connection refused - webui not yet started
                     continue
                 else:
+                    self._take_vm_screenshot('screenshot_webui_not_accessible.ppm')
                     raise e
             else:
                 break
 
+    def _take_vm_screenshot(self, name):
+        """ Creates screenshot of VMs screeen
+
+        :param name: screenshot filename
+        :type name: str
+        """
+        screenshot_file = os.path.join(
```

Comment:
please use temporary file for the screenshot and then just call `self.addLog` with `copy_file=True`. That way, you don't need to deal with the file paths in any way.

Reply:
I was assuming virsh wouldn't like it, thanks